packages/api/src/controllers/room.ts

```diff
@@ -17,18 +21,119 @@ app.post("/", authorizer({}), async (req, res) => {
     req.config.livekitApiKey,
     req.config.livekitSecret
   );
-  const opts = {
+  const room = await svc.createRoom({
     name: id,
     // timeout in seconds
-    emptyTimeout: 30 * 60,
-    maxParticipants: 2,
-  };
-  const room = await svc.createRoom(opts);
+    emptyTimeout: 15 * 60,
+    maxParticipants: 10,
+    metadata: `{"userId": "${req.user.id}"}`,
+  });
   console.log("livekit room created", room);
+
+  await db.room.create({
+    id: id,
+    userId: req.user.id,
+    participants: {},
+    events: [],
+  });
+
   res.status(201);
   res.json({
     id: id,
   });
 });
 
+app.get("/:roomId", authorizer({}), async (req, res) => {
+  const room = await db.room.get(req.params.roomId);
+
+  if (!req.user.admin && req.user.id !== room.userId) {
+    throw new ForbiddenError(`users may only view their own rooms`);
+  }
+
+  res.status(200).json(toExternalRoom(room, req.user.admin));
+});
+
+function toExternalRoom(room: Room, isAdmin = false) {
+  if (isAdmin) {
+    return room;
+  }
+  room.userId = undefined;
+  for (const key in room.participants) {
+    room.participants[key].tracksPublished = undefined;
+  }
```

Comment:
Maybe consider hiding the entire `participants` field. Do users really need access to all that information? Might be easier to start by using it only internally for tracking/billing and consider exposing only parts of it that users may need.

Reply:
I think it will be helpful to give them at least the IDs and names of the participants so that they would easily be able to see who has joined and remove people if needed (api coming in next PR). I will tidy up the remaining fields too in the next PR.